src/filter.js

```diff
@@ -36,12 +35,11 @@ var keys = require('./keys');
 module.exports = _curry2(_dispatchable('filter', _xfilter, function(pred, filterable) {
   return (
     _isObject(filterable) ?
-      _reduce(function(acc, key) {
+      _accumObj(function(key, value, newObject) {
         if (pred(filterable[key])) {
-          acc[key] = filterable[key];
+          newObject[key] = value;
         }
-        return acc;
-      }, {}, keys(filterable)) :
+      }, filterable) :
     // else
```

Comment:
Or you could just use the native `trim()` function even though it's not supported in IE8.